File: cakk-api/src/main/java/com/cakk/api/aspect/AopForTransaction.java
package com.cakk.api.aspect;

import org.aspectj.lang.ProceedingJoinPoint;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

@Component
public class AopForTransaction {

	@Transactional(propagation = Propagation.REQUIRES_NEW)
	public Object proceed(final ProceedingJoinPoint joinPoint) {
		try {
			return joinPoint.proceed();
		} catch (Throwable e) {
			throw new RuntimeException(e);
		}
	}
}

File: cakk-core/src/main/kotlin/com/cakk/core/aspect/AopForTransaction.kt
package com.cakk.api.aspect;

import org.aspectj.lang.ProceedingJoinPoint;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

import com.cakk.common.exception.CakkException;

@Component
public class AopForTransaction {

	@Transactional(propagation = Propagation.REQUIRES_NEW)
	public Object proceed(final ProceedingJoinPoint joinPoint) {
		try {
			return joinPoint.proceed();
		} catch (Throwable e) {
			if (e instanceof CakkException) {
				throw (CakkException) e;
			}

			throw new RuntimeException(e);
		}
	}
}
